### PR TITLE
DEPR: positional indexing on Series __getitem__/__setitem__

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

xref #50617, possibly closes

### CODE REVIEW

File: pandas/tests/series/methods/test_values.py [hunk 1]
```diff
@@ -25,5 +25,5 @@ def test_values_object_extension_dtypes(self, data):
 
     def test_values(self, datetime_series):
         tm.assert_almost_equal(
-            datetime_series.values, datetime_series, check_dtype=False
+            datetime_series.values, list(datetime_series), check_dtype=False
```
Comment (rhshadrach) [inline pandas/tests/series/methods/test_values.py:28]: Curious why this was necessary

**Review by rhshadrach — COMMENTED** [review]
lgtm

File: pandas/tests/series/methods/test_values.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/series/methods/test_values.py:28]: IIRC bc assert_almost_equal does

```
        for i in range(len(a)):
            try:
                assert_almost_equal(a[i], b[i], rtol=rtol, atol=atol)
```

Comment (rhshadrach) [reply pandas/tests/series/methods/test_values.py:28]: Should we be using iloc in assert_almost_equal instead?

**Review by rhshadrach — APPROVED** [review]
lgtm

File: pandas/tests/series/indexing/test_get.py [hunk 2]
```diff
@@ -144,17 +144,24 @@ def test_get_with_default():
...
+
+            assert s.get("e", "z") == "z"
+            assert s.get("e", "e") == "e"
+
+            msg = "Series.__getitem__ treating keys as positions is deprecated"
+            warn = None
+            if index is d0:
+                warn = FutureWarning
+            with tm.assert_produces_warning(warn, match=msg):
+                assert s.get(10, "z") == "z"
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_get.py:163]: NDFrame.get is a try/except around `__getitem__`.  xref #17928

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/tests/series/indexing/test_get.py:163]: Ah that's annoying. Okay makes sense here

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**